app/models/manageiq/consumption/showback_price_plan.rb

```diff
@@ -14,19 +14,21 @@ class ManageIQ::Consumption::ShowbackPricePlan < ApplicationRecord
   # Returns the total accumulated costs for all rates that apply
   ###################################################################
   def calculate_total_cost(event)
+    calculate_total_cost_input(event.resource.type, event.data, event.time_span, event.month_duration, event.context)
+  end
+
+  def calculate_total_cost_input(category, data, time_span, cycle_duration, context)
```

Comment:
Maybe category is not a good name when is the resource type , Should we change it?